comp/trace/agent/impl/run.go

```diff
@@ -70,6 +71,20 @@ func runAgentSidekicks(ag component) error {
 	} else {
 		ag.Agent.DebugServer.AddRoute("/config", ag.config.GetConfigHandler())
 	}
+	// Adding a route to trigger a secrets refresh from the CLI.
+	// TODO - components: the secrets comp already export a route but it requires the API component which is not
+	// used by the trace agent. This should be remove once the trace-agent is fully componentize.
+	ag.Agent.DebugServer.AddRoute("/secret/refresh", http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
```

Comment:
we probably should validate the auth token is set in the request with `apiutil.Validate(w, req)`?